core/types/receipt.go

```diff
@@ -84,14 +84,16 @@ type Receipt struct {
 	BlockNumber      *big.Int    `json:"blockNumber,omitempty"`
 	TransactionIndex uint        `json:"transactionIndex"`
 
-	// Optimism: extend receipts with L1 fee info
+	// Optimism: extend receipts with L1 and operator fee info
 	L1GasPrice          *big.Int   `json:"l1GasPrice,omitempty"`          // Present from pre-bedrock. L1 Basefee after Bedrock
 	L1BlobBaseFee       *big.Int   `json:"l1BlobBaseFee,omitempty"`       // Always nil prior to the Ecotone hardfork
 	L1GasUsed           *big.Int   `json:"l1GasUsed,omitempty"`           // Present from pre-bedrock, deprecated as of Fjord
 	L1Fee               *big.Int   `json:"l1Fee,omitempty"`               // Present from pre-bedrock
 	FeeScalar           *big.Float `json:"l1FeeScalar,omitempty"`         // Present from pre-bedrock to Ecotone. Nil after Ecotone
 	L1BaseFeeScalar     *uint64    `json:"l1BaseFeeScalar,omitempty"`     // Always nil prior to the Ecotone hardfork
 	L1BlobBaseFeeScalar *uint64    `json:"l1BlobBaseFeeScalar,omitempty"` // Always nil prior to the Ecotone hardfork
+	OperatorFeeScalar   *uint64    `json:"operatorFeeScalar,omitempty"`   // Always nil prior to the Isthmus hardfork
+	OperatorFeeConstant *uint64    `json:"operatorFeeConstant,omitempty"` // Always nil prior to the Isthmus hardfork
```

Comment:
If these are set from Isthmus forward I think it will break compatibility with tools like cast (and geth's receipt parsing) as they really don't like unknown fields. Not a major issue if they're only set when non-zero since it would only affect chains that actually use this, but could be quite problematic if it applies to all chains.

Reply:
How did we deal with this when the L1 fee info was added?

AFAICT the fields will be set for all chains which activate Isthmus, even if the fields are zero.

I see a few ways forward:

a) Hide the fields in the receipt if they are zero. This is going to add a little complexity to the receipt marshalling code and divergence from how we are handling the L1 fee information in the receipt. 
b) Have the parameters default to `nil` pointers until they are explicitly set. This simplifies receipt marshalling but at the expense of more complexity (nullity checks) in the core logic
c) Lean into having the new parameters in receipts on all chains,  and try to modify tools which consume these receipts to be able to handle the new fields.

Reply:
I've done solution b) (leave those fields empty)

Reply:
This solution looks to me good now, thanks. The API parsing code looks correctly adapted to omit nil fields.